src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -5957,7 +5957,11 @@ internal bool DeserializeUnencryptedValue(SqlBuffer value, byte[] unencryptedByt
                             unencryptedBytes = bytes;
                         }
 
-                        value.SqlBinary = new SqlBinary(unencryptedBytes);   // doesn't copy the byte array
+#if NET7_0_OR_GREATER
+                        value.SqlBinary = SqlBinary.WrapBytes(unencryptedBytes);
+#else
+                        value.SqlBinary = SqlTypeWorkarounds.SqlBinaryCtor(unencryptedBytes, true);   // doesn't copy the byte array
+#endif
```

Comment:
This may not be safe because unencryptedBytes is a parameter so you will need to audit all the callsites to make sure that ownership is always relinquished by the callers.